src/main/java/net/gcardone/junidecode/Junidecode.java

```diff
@@ -627,9 +642,61 @@ public static String unidecode(final String s) {
              * one is reserved.
              */
             if (low < map.length) {
-                sb.append(map[low]);
+                caseConversion.append(sb, map[low]);
```

Comment:
I don't think that the conversion belongs on this line, because it causes a call to a function for each codepoint. It would be more efficient to do it once, for the whole string.

I suggest to implement this functionality inverting the dependency: `unidecode(final String s, final Case caseConversion)` should call `unidecode(final String s)` _and then_ apply the case conversion on the entire output. What do you think?

Reply:
I do not agree here as Java internally is designed very inefficient: A `String` is a wrapper for an array of characters (`char[]`). If you do `toLowerCase` on that, you have to create a new `char[]` and a new `String` wrapping it and again looping over every character to do the conversion. In that case, there would be no need at all to add this use-case to your library as then I can even just call `toLowerCase(Locale.ROOT)` you the result of junidecode. However, this is very inefficient if you process lots of text (especially since java does not know that we already have ASCII here and do not need to care about the complexity of case conversion with full unicode support).
If you care about the extra method call (that `JIT` will optimize away), I can remove the `ORIGINAL_CASE` and supply `null` instead so that using an `if (caseConversion==null)` the original code would still apply without the extra method call wrapping it.

Reply:
I updated the PR accordingly. Now it is guaranteed (even when running your code in GraalVM or TeaVM) that no extra Method call will happen if junidecode is used as before. The extra `if` will be optimized away by JIT as `caseConversion` is final and so if it once was `null` all further loop runs will produce the same result so no overhead for subsequent loopings. (If you would still be not convinced I would recommend to run some performance tests as otherwise, you would have to duplicate the code of the method and IMHO that would not be a good decision for maintenance reasons).

Reply:
No reaction? Do have to create my own fork of this project?

Reply:
Thanks, I much prefer the new version.

Sorry for the delay. On the bright side I refreshed my credentials with sonatype, so once your change is merged I can quickly release a new jar.